stream-chat-android-core/src/main/java/io/getstream/chat/android/client/call/CoroutineCall.kt

```diff
@@ -12,17 +12,23 @@ import kotlinx.coroutines.withContext
 @InternalStreamChatApi
 public class CoroutineCall<T : Any>(
     private val scope: CoroutineScope,
-    public val suspendingTask: suspend () -> Result<T>,
+    private val suspendingTask: suspend CoroutineScope.() -> Result<T>,
 ) : Call<T> {
 
     private var job: Job? = null
 
+    internal suspend fun awaitImpl(): Result<T> {
+        return withContext(scope.coroutineContext) {
+            suspendingTask()
+        }
+    }
+
     override fun cancel() {
         job?.cancel()
     }
 
     override fun execute(): Result<T> {
-        return runBlocking { suspendingTask() }
+        return runBlocking(block = suspendingTask)
```

Comment:
Ended up not specifying the coroutine context here, because doing so makes tests fail when we're calling nested suspending Room Dao methods inside a transaction.